apps/prairielearn/src/lib/question-variant.ts

```diff
@@ -180,26 +195,100 @@ async function makeAndInsertVariant(
     question_course,
     options,
   );
-  const variant = await sqldb.callRow(
-    'variants_insert',
-    [
-      variantData.variant_seed,
-      variantData.params,
-      variantData.true_answer,
-      variantData.options,
-      variantData.broken,
-      instance_question_id,
-      question.id,
-      course_instance_id,
-      user_id,
-      authn_user_id,
-      group_work,
-      require_open,
-      variant_course.id,
-      client_fingerprint_id,
-    ],
-    VariantWithFormattedDateSchema,
-  );
+
+  const variant = await sqldb.runInTransactionAsync(async () => {
```

Comment:
I can't comment on the line, but: the `group_work` parameter to `makeAndInsertVariant` is now unused (and was in fact unused before - we passed it to the sproc, but the sproc did nothing with it). Consider removing to clean that up.

Reply:
Done in 79ff2ab9e1e4c0b6b3f0bb2e15c3beb9fa466f47. This actually cascaded into multiple functions that also only had that parameter to pass it along. I'll do another passthrough to ensure everything is fine before merging.

Reply:
It was 5 functions, most of them local, so not too much.